src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -441,6 +443,212 @@ public List<PackageManagerSearchElement> ListAll()
             return CachedPackageList;
         }
 
+        internal void InitiatePackageDownloadAndInstall(IPackageInfo packageInfo, string downloadPath = null)
+        {
+            var header = Model.GetPackageHeader(packageInfo);
+            if (header == null)
+            {
+                // TODO: move message to resources
+                var result = MessageBox.Show(string.Format("{0} could not be downloaded.", packageInfo.Name),
+                "Download Error", MessageBoxButton.OKCancel, MessageBoxImage.Question);
+                return;
+            }
+
+            var version = Model.GetGregPackageVersion(header, packageInfo.Version);
+            if (version == null)
+            {
+                // TODO: move message to resources
+                var result = MessageBox.Show(string.Format("Version {0} of {1} could not be found.", packageInfo.Version.ToString(), packageInfo.Name),
```

Comment:
Do you still intend to finish the TODO in this PR?

Reply:
Yes, thanks for pointing this out. I have moved them now